### PR TITLE
Djuun

### PR DESCRIPTION
**Pull request by benjub**
Mainly:
* big shortening of djuun
* restructuring, with three subsections
* theorems with label ending by 'N' are meant to ultimately replace the similar labels without an 'N' (this entails minimal modifications of a few proofs). This globally reduces proof size (not by a small margin), and it states most theorems abount injections using e.g. ``( inl |` A )`` instead of ``inl``. As I explain in a head comment, `inl` has to be thought as a "template" for injections, and the meaningful objects are actually the functions ``( inl |` A ) : A --> (A |_| B )``

I would like to merge this now to ease further modifications. However, if you think it's not mature enough to be merged, then that's fine and I'll remove all the xxxN/xxx before merging.

### CODE REVIEW

File: iset.mm
Comment (jkingdon) [inline iset.mm:63902]: This (and the other results involving restriction) make sense to me, I had been thinking of maybe doing the same myself.

**Review by jkingdon — APPROVED** [review]

**Comment by jkingdon** [comment]
> I would like to merge this now to ease further modifications. However, if you think it's not mature enough to be merged, then that's fine and I'll remove all the xxxN/xxx before merging.

Whichever will get djudom done faster &lt;wink&gt;.

More seriously, if anyone else wants to weigh in, go for it, but as far as I'm concerned you can merge this now if you'd like.

**Comment by benjub** [comment]
A few more shortenings in the same vein, including some large ones (the proof of djuunN is now much shorter than that of djuun, does not use djur, and is setvar-free!).
I propose to replace djur with eldju, too (which is a biconditional and involves the restricted injections). And even that should be less used than djuunN. For those results, setvar-free proofs are possible (reflected in the fact that these constructions work in category theory, where objects need not have elements).

I'll do the removals in a separate PR.

Side remark about djuinN/djuunN versus the old ones: since ``( inl |` A )`` is the more natural object (rather than `inl`), it makes more sense (and is faster to prove) the versions with ``ran ( inl |` A )`` rather than `( inl " A )`. (Sorry for having used the latter when informally stating the main results expected from such a section, in the initial Issue.)

**Comment by jkingdon** [comment]
> I propose to replace djur with eldju, too (which is a biconditional and involves the restricted injections)

Should be fine, but maybe hold up on removing theorems until #2659 can be completed and merged? One of the proofs there uses djur and although revising it to use eldju (or djuun if feasible) is certainly an option at some point, my instinct would be to get some version of it working before we go too wild.

**Pull request merged**
